pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginManager.java

```diff
@@ -41,10 +51,18 @@ public class PluginManager {
 
   public static final String PLUGINS_DIR_PROPERTY_NAME = "plugins.dir";
   public static final String PLUGINS_INCLUDE_PROPERTY_NAME = "plugins.include";
+  public static final String PLUGINS_LOADER_LEGACY_PROPERTY_NAME = "pinot.plugins.loader.legacy";
+  public static final String PLUGINS_LOADER_LEGACY_DEFAULT = "false";
```

Comment:
This should be true for now.

Reply:
Probably we should rename the property to be something like `pinot.plugins.loader=plexus`. In case the property is not defined, we use the legacy mode. In the future we will be able to change the default to plexus.

Reply:
If we make legacy the default, I wonder if anyone dares to switch to plexus classworld, even though classloader issues keep happening. Instead I propose to make plexus classworlds behave more like its current behavior. i.e. having access to all classes (like now) but loading them in the expected order. In other words: make them unlimited. The switch to limited as default is smaller.

Reply:
We are used to use flags to enable new features to be sure we do not affect deployments where the flag is not set. I'm sure StarTree will use the new system in their deployments (after testing it for a while), so I'm sure it is going to be used.